edb/common/struct.py

```diff
@@ -347,14 +359,6 @@ def _init_fields(
 
             setattr(self, field_name, value)
 
-    def __setattr__(self, name: str, value: Any) -> None:
```

Comment:
I'm wondering if we can speed up `_init_fields` by not calling `_getdefault` and instead implementing `__getattr__` to call it dynamically when needed.

Reply:
Overloading `__getattr__` has a steep perf penalty even if it's very simple.  It also kills typing checks.

Reply:
> Overloading __getattr__ has a steep perf penalty even if it's very simple. 


Why? It would only be called for slots that weren't initialized via the constructor. Essentially lazy defaults. You can even cache defaults (write them in) when `__getattr__` is invoked on them.

> It also kills typing checks.

Why? Literally nothing changes.

Reply:
> Why? Literally nothing changes.

Mypy stops typechecking attribute accesses when there is a `__getattr__` on a class.